pkg/dogstatsd/enrich.go

```diff
@@ -40,26 +41,36 @@ func enrichTags(tags []string, defaultHostname string, originTagsFunc func() []s
 		}
 	}
 	tags = tags[:n]
+
 	if entityIDValue == "" || !entityIDPrecedenceEnabled {
 		// Add origin tags only if the entity id tags is not provided
-		tags = append(tags, originTagsFunc()...)
+		originTags = originTagsFunc()
 	}
 	if entityIDValue != "" && entityIDValue != entityIDIgnoreValue {
 		// Check if the value is not "none" in order to avoid calling
 		// the tagger for entity that doesn't exist.
 
 		// currently only supported for pods
 		entity := kubelet.KubePodTaggerEntityPrefix + entityIDValue
-		entityTags, err := getTags(entity, tagger.DogstatsdCardinality)
+		var err error
+		entityTags, err = getTags(entity, tagger.DogstatsdCardinality)
 		if err != nil {
 			log.Tracef("Cannot get tags for entity %s: %s", entity, err)
 			tlmUDPOriginDetectionError.Inc()
-		} else {
-			tags = append(tags, entityTags...)
 		}
 	}
-	return tags, host
+
+	if len(originTags) == 0 && len(entityTags) == 0 {
+		return tags, host
+	}
+
+	finalTags := make([]string, 0, len(tags)+len(originTags)+len(entityTags))
+	finalTags = append(finalTags, tags...)
+	finalTags = append(finalTags, originTags...)
+	finalTags = append(finalTags, entityTags...)
```

Comment:
actually, you could do this only if `cap(tags) < len(tags)+len(originTags)+len(entityTags)`, otherwise just append to existing tags slice

Reply:
Good idea, implemented